test/e2e/alerts-existing-cozy.js

```diff
@@ -0,0 +1,223 @@
+    groups: allGroups
+  } = await fetchData(client)
+
+  const notificationSettings = allSettings.find(
+    setting => setting.notifications
+  )
+  const balanceLowerRules = notificationSettings.notifications.balanceLower
+
+  const accountsById = keyBy(allAccounts, getDocumentID)
+  const groupsById = keyBy(allGroups, getDocumentID)
+  const formatAccountOrGroup = accountOrGroupFormatter(accountsById, groupsById)
+  const formatRule = ruleFormatter(formatAccountOrGroup)
+
+  console.log('Existing rules:')
+  console.log(
+    balanceLowerRules
+      .map(formatRule)
+      .map(x => `- ${x}`)
+      .join('\n')
+  )
+
+  for (const rule of balanceLowerRules) {
+    const correspondingAccounts = allAccounts.filter(
+      ruleAccountFilter(rule, allGroups)
+    )
+    let chosenAccountId
+    if (correspondingAccounts.length > 1) {
+      console.log('More than 1 corrsponding account for rule', formatRule(rule))
+      chosenAccountId = (await chooseAccount(correspondingAccounts))._id
+    } else if (correspondingAccounts.length === 1) {
+      chosenAccountId = correspondingAccounts[0]._id
+    } else {
+      console.warn('No account for rule', formatRule(rule), '. Bailing out.')
+      continue
+    }
+
+    const SNACKS_CATEGORY_ID = '400160'
+    const transaction = {
+      label: `Fake transaction for ${formatRule(rule)}`,
+      amount: -9,
+      account: chosenAccountId,
+      _type: TRANSACTION_DOCTYPE,
+      manualCategoryId: SNACKS_CATEGORY_ID,
+      date: new Date().toISOString(),
+      cozyMetadata: {
+        createdByApp: CREATED_BY_APP_ID
+      }
+    }
+
+    console.log('Creating transaction', {
+      transaction: formatTransaction(transaction),
+      rule: formatRule(rule),
+      account: formatAccountOrGroup(accountsById[chosenAccountId])
+    })
+    console.log('Saving transaction...')
+    const { data: newTransaction } = await client.save(transaction)
+
+    // We have to fake an update to wake up the onOperationOrBillCreate service
+    console.log('Updating transaction...')
+    await client.save({
+      ...newTransaction,
+      amount: -10
+    })
+  }
+  console.log('Saving fake bill')
+  const { data: newBill } = await client.save({
+    _type: BILLS_DOCTYPE,
+    cozyMetadata: {
+      createdByApp: CREATED_BY_APP_ID
+    },
+    amount: 0,
+    vendor: CREATED_BY_APP_ID,
+    label: 'Fake bill to wake up onOperationOrBillCreate'
+  })
+  console.log(newBill)
+}
+
+const cleanupFakeDocumentsForDoctype = async (client, doctype) => {
+  console.log(`Fetching fake ${doctype}...`)
+  const { data: fakeDocuments } = await client.query({
+    doctype: doctype,
+    selector: {
+      'cozyMetadata.createdByApp': CREATED_BY_APP_ID
+    }
+  })
+  console.log(`Found ${fakeDocuments.length} fake ${doctype}`)
+  console.log(`Cleaning up ${fakeDocuments.length} fake ${doctype}...`)
+  const resp = await client.stackClient
+    .collection(doctype)
+    .destroyAll(fakeDocuments.map(x => omit(x, '_type')))
```

Comment:
Shouldn't `destroyAll` do the `omit`? It's weird to have to remove a property on an object we got from a cozy-client's query in order to destroy it.

Reply:
Very good point: https://github.com/cozy/cozy-client/pull/597, I had the same hunch but was too lazy to fix it right away ;)